app/lib/widgets/settings/share_settings.dart

```diff
@@ -0,0 +1,235 @@
+  Widget build(BuildContext context) {
+    return SafeArea(
+      child: Column(
+        children: [
+          _Section(
+            title: 'Share ${widget.bike.name}',
+            children: [
+              Form(
+                key: _formKey,
+                autovalidateMode: AutovalidateMode.always,
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  crossAxisAlignment: CrossAxisAlignment.stretch,
+                  children: [
+                    Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        TextFormField(
+                          decoration: const InputDecoration(
+                            icon: Icon(Icons.email),
+                            hintText: 'VanMoof account email',
+                            labelText: 'Email',
+                          ),
+                          onSaved: (value) {
+                            _email = value ?? '';
+                          },
+                          validator: (String? value) {
+                            return (value != null &&
+                                    !EmailValidator.validate(value))
+                                ? 'Not a valid email address'
+                                : null;
+                          },
+                        ),
+                        Text("${_duration.round()} days"),
+                        CupertinoSlider(
+                          value: _duration,
+                          min: 1.0,
+                          max: 365.0,
+                          onChanged: (newDuration) {
+                            setState(() => _duration = newDuration);
+                          },
+                        ),
+                        Padding(
+                          padding: const EdgeInsets.symmetric(vertical: 6),
+                          child: ElevatedButton(
+                            onPressed: () => submitForm(Navigator.of(context)),
+                            child: const Text('Share'),
+                          ),
+                        ),
+                        error != null
+                            ? Text(
+                                error!,
+                                style: TextStyle(
+                                    color: Theme.of(context).errorColor),
+                              )
+                            : Container(),
+                        success != null
+                            ? Text(
+                                success!,
+                                style: const TextStyle(color: Colors.green),
+                              )
+                            : Container(),
+                      ],
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _ShareHolderList extends State<ShareSettings> {
```

Comment:
I think we should change the way this class works.

1. We should not store a Promise that might resolves to a list of share holders, instead we should await the `getCurrentShares` promise and store it's result in `_shareHolders` and call `setState` so the UI updates.
```
  dynamic _shareHolders;

  obtainShareHolders() async {
    _shareHolders = await api?.getCurrentShares(widget.bike.id);
    setState(() {});
  }

  @override
  void initState() {
    super.initState();
    obtainShareHolders();
  }
```

2. Now you can remove the `FutureBuilder` and in the build method just check if `_shareHolders` is not `null`. If it's `null` it's still loading, if it isn't null the value is set and thus we can show the list.